Removing hard-coded logging configuration

jluethi opened the pull request:
Using logging.basicConfig at this level of the program leads to this logging configuration being forced upon all programs that import jgo. Seeing how useful jgo is as a library, it should follow the python logging convention of libraries not hard-coding the logging configuration: https://docs.python.org/3/howto/logging.html#configuring-logging-for-a-library

----

Example code:
```
import logging
logging.basicConfig(filename='example.log', level=logging.DEBUG)
logger = logging.getLogger(__name__)
print(logger.name)
logger.debug('This message should go to the log file')
logger.info('So should this')
logger.warning('And this, too')
```
This code logs a few example messages to a log file. No library import should change how this code works. Adding `import jgo` at the beginning (or importing any library with `jgo` as a dependency, [see issue here](https://github.com/imagej/pyimagej/issues/40)) changes the behavior of the logger, such that the logging level is set to INFO and logging is sent to the console, not the file. The reason for this is the hard-coded logging.basicConfig in the jgo library. Removing lines 29-33 from jgo.jgo.py fixes this issue.

-----

I tested that this does not interfere with using the jgo library via pyimagej. If anything relies on jgo logging (hopefully not), that would need to be adapted. If other libraries did not initialize the logging and relied on the jgo initialization, logging.INFO messages wouldn't be automatically displayed anymore. Warnings would still be printed to the console if no logging is set up.

ctrueden commented:
Wow, thanks for the detective work, @jluethi!

@hanslovsky What do you think?

hanslovsky commented:
I absolutely agree, thanks for raising this @jluethi!

It would probably good to run `logging.basicConfig` at the beginning of [`jgo_main`](https://github.com/scijava/jgo/blob/master/jgo/jgo.py#L252). As far as I understand, this should not affect anything else because it is scoped only within the main method and should not be executed when `jgo` is imported (but only when `jgo_main` is run). What do you think @jluethi @ctrueden?

Besides that, absolutely :+1: to merge from my side!

ctrueden commented:
> It would probably good to run logging.basicConfig at the beginning of jgo_main.

@hanslovsky Can we think of any case where someone would want to call `jgo_main` as API in other contexts? And if so, whether dictating the logging would create any problems? If not, and the only intended use of `jgo_main` is an main program entry point, then I agree with you.

@jluethi Could you please migrate the basic logging configuration to `jgo_main` as another commit on ths PR? Then we can merge it.

jluethi commented:
@ctrueden @hanslovsky Sounds good to me. I added the logging functionality to the jgo_main function. When I test it, it does not interfere with the logging in my program. Thus, I updated the PR to include logging in jgo_main.

I don't have a test case to check whether the logging now works as intended for directly calling jgo_main. @hanslovsky Can you test this?

I don't know if anyone would directly use jgo_main. Looking on its usage on public github repositories, it seems to be called by @hanslovsky conda-build-scripts, e.g. [here](https://github.com/hanslovsky/conda-build-scripts/blob/2626d72dd1a88d4729b876696337ed20349160a7/paintera/paintera/paintera.py#L46), but jgo_main is not found anywhere else in public repositories (using github search)

ctrueden commented:
Thanks, @jluethi!

> I don't know if anyone would directly use jgo_main. Looking on its
> usage on public github repositories, it seems to be called by
> @hanslovsky conda-build-scripts, e.g. here, but jgo_main is not
> found anywhere else in public repositories (using github search)

I think its main use is from the [shell script hook that gets installed](https://github.com/scijava/jgo/blob/4bac7a0d4b4cf5a8aa4cee9340e6ec83d1eeaf4c/setup.py#L21-L25). That allows you to use `jgo` on the command line, which launches Python and runs the `jgo_main` function.

I'll merge now. :+1:

ctrueden commented:
I released 0.4.0 (shamefully, without testing). Let us know if it solves your problem.

jluethi commented:
@ctrueden Wow, that was a super fast release! Thanks a lot! 👏 
And yes, it solves the logging issue for me 👍

hanslovsky commented:
Sorry for the late response: I pretty much agree with everything that @ctrueden and @jluethi discussed. `jgo_main` is indeed ment to be used only internally and through the console endpoint. I will rename it to `_jgo_main` to indicate that intent as is convention in the Python world.
